### PR TITLE
NodeJS Sample code with Test for Azure Functions

### PR DESCRIPTION
**Pull request by issacnitin**
Please review after merging #124

### CODE REVIEW

File: node/plain/functionApp/Application/Function1/sample.dat [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "name": "Azure"
```
Comment (sachinma) [inline node/plain/functionApp/Application/Function1/sample.dat]: why is this file required?

File: node/plain/functionApp/ArmTemplates/windows-functionapp-template.json [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                        }
+                    },
+                    "keySource": "Microsoft.Storage"
+                }
+            },
+            "dependsOn": []
+        },
+      {
+        "type": "Microsoft.Web/serverfarms",
+        "sku": {
```
Comment (sachinma) [inline node/plain/functionApp/ArmTemplates/windows-functionapp-template.json]: aren't you using consumption plan?

File: node/plain/functionApp/ArmTemplates/windows-functionapp-template.json [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        "apiVersion": "2018-05-01-preview",
+        "name": "[parameters('functionAppName')]",
+        "type": "microsoft.insights/components",
+        "location": "[resourceGroup().location]",
+        "tags": {
+          "[concat('hidden-link:', resourceGroup().id, '/providers/Microsoft.Web/sites/', parameters('functionAppName'))]": "Resource"
+        },
+        "properties": {
+          "ApplicationId": "[parameters('functionAppName')]",
+          "Request_Source": "IbizaWebAppExtensionCreate"
```
Comment (sachinma) [inline node/plain/functionApp/ArmTemplates/windows-functionapp-template.json]: change it to IbizaDevOpsExtensionCreate

File: node/plain/functionApp/Tests/index.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+var tests = require('require-dir')('./unittests');
+
+for (var functions in tests) {
```
Comment (sachinma) [inline node/plain/functionApp/Tests/index.js]: run it using gulp/grunt tasks - since they would publish the results to DevOps server

File: node/plain/functionApp/Application/Function1/sample.dat (hunk 1 above)
Comment (issacnitin) [reply node/plain/functionApp/Application/Function1/sample.dat]: It's generated by VSCode, I was not sure if this file should be removed.. Function works perfectly without it, so I'll remove it for now

File: node/plain/functionApp/ArmTemplates/windows-functionapp-template.json (hunk 2 above)
Comment (issacnitin) [reply node/plain/functionApp/ArmTemplates/windows-functionapp-template.json]: Yes, consumption plan is the default

File: node/plain/functionApp/Application/Function1/sample.dat (hunk 1 above)
Comment (sachinma) [reply node/plain/functionApp/Application/Function1/sample.dat]: lets leave it in that case

File: node/plain/functionApp/Tests/index.js (hunk 4 above)
Comment (issacnitin) [reply node/plain/functionApp/Tests/index.js]: Done, I've used gulp

File: node/plain/functionApp/ArmTemplates/windows-functionapp-template.json (hunk 3 above)
Comment (issacnitin) [reply node/plain/functionApp/ArmTemplates/windows-functionapp-template.json]: Done
